test/e2e/aws_test.go

```diff
@@ -48,49 +59,75 @@ func (a *awsTestHelper) makeCredentialsSecret(namespace string) *corev1.Secret {
 	}
 }
 
+func (a *awsTestHelper) defaultExternalDNS(credsSecret *corev1.Secret) operatorv1alpha1.ExternalDNS {
+	return operatorv1alpha1.ExternalDNS{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testExtDNSName,
+		},
+		Spec: operatorv1alpha1.ExternalDNSSpec{
+			Zones: []string{hostedZoneID},
+			Source: operatorv1alpha1.ExternalDNSSource{
+				ExternalDNSSourceUnion: operatorv1alpha1.ExternalDNSSourceUnion{
+					Type: operatorv1alpha1.SourceTypeService,
+					Service: &operatorv1alpha1.ExternalDNSServiceSourceOptions{
+						ServiceType: []corev1.ServiceType{
+							corev1.ServiceTypeLoadBalancer,
+							corev1.ServiceTypeClusterIP,
+						},
+					},
+					AnnotationFilter: map[string]string{
+						"external-dns.mydomain.org/publish": "yes",
+					},
+				},
+				HostnameAnnotationPolicy: "Ignore",
+				FQDNTemplate:             []string{fmt.Sprintf("{{.Name}}.%s", hostedZoneDomain)},
+			},
+			Provider: operatorv1alpha1.ExternalDNSProvider{
+				Type: operatorv1alpha1.ProviderTypeAWS,
+				AWS: &operatorv1alpha1.ExternalDNSAWSProviderOptions{
+					Credentials: operatorv1alpha1.SecretReference{
+						Name: credsSecret.Name,
+					},
+				},
+			},
+		},
+	}
```

Comment:
The same structure is repeated in each provider which multiples the potential mistakes done in the definition of ExternalDNS instance. And it's important that all the providers use the same specification of this custom resource as they share the same tests.
I'm proposing to keep `defaultExternalDNS` function in `utils.go` and call it from all providers adding the provider specific things after:
```go
func (a *awsTestHelper) externalDNS(credsSecret *corev1.Secret) operatorv1alpha1.ExternalDNS {
    extDNS := defaultExternalDNS()
    extDNS.Provider = operatorv1alpha1.ExternalDNSProvider{
        Type: operatorv1alpha1.ProviderTypeAWS,
            AWS: &operatorv1alpha1.ExternalDNSAWSProviderOptions{
	        Credentials: operatorv1alpha1.SecretReference{
			Name: credsSecret.Name,
	        },
            },
      }
      return extDNS
}
```

Reply:
agreed with you, is there any chance we get the operator structure differ based on provider?